assets/src/edit-story/components/postLock/postLock.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useFeatures } from 'flagged';
+import { trackError } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../app/api';
+import { useStory } from '../../app/story';
+import { useConfig } from '../../app/config';
+import PostLockDialog from './postLockDialog';
+
+function PostLock() {
+  const {
+    actions: { getStoryLockById, setStoryLockById, deleteStoryLockById },
+  } = useAPI();
+  const {
+    storyId,
+    userId,
+    dashboardLink,
+    postLock: { interval: postLockInterval, showLockedDialog },
+  } = useConfig();
+
+  const { previewLink } = useStory(({ state: { story: { previewLink } } }) => ({
+    previewLink,
+  }));
+  const [showDialog, setShowDialog] = useState(false);
+  const [user, setUser] = useState({});
+  const { enablePostLocking } = useFeatures();
+
+  const closeDialog = useCallback(() => {
+    setShowDialog(false);
+    setStoryLockById(storyId);
+  }, [setStoryLockById, storyId]);
+
+  const doGetStoryLock = useCallback(() => {
+    if (enablePostLocking && showLockedDialog) {
+      getStoryLockById(storyId)
+        .then((result) => {
+          if (result.locked && result.user !== userId) {
```

Comment:
With my suggestion below to simplify the REST API response, we don't even need to compare with `userId` here. If `result.user` is set, it will never be the current user.

Reply:
I don't like making REST API depend on current user ID for context. This is not how the WP's REST API works and I don' think it should work that way here. It is a bit of a pain to check this locally, but it much more RESTFUL that way and more generic.